tensorflow-keras/src/main/java/org/tensorflow/keras/utils/SymbolicShape.java

```diff
@@ -0,0 +1,92 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+=======================================================================*/
+package org.tensorflow.keras.utils;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.tensorflow.Operand;
+
+/**
+ * Utility class that handles sybmolic shapes so that the shapes can be resolved during runtime and
+ * stay consistent across operands.
+ *
+ * <p>The same symbols accros variaous shapes should have the same dimension values.
```

Comment:
Typos.

Reply:
OK

Reply:
I removed SymbolicShapes from this PR as it is needed when I do a PR on metrics.